app/src/features/Preview/Preview.tsx

```diff
@@ -0,0 +1,215 @@
+    attributeId?: string;
+  }>();
+  const { data: mapping } = useApiResourcesRetrieveQuery(
+    {
+      id: mappingId ?? "",
+    },
+    {
+      skip: !Boolean(mappingId),
+    }
+  );
+  const [fhirInstance, setFhirInstance] = useState<
+    typeof patientJson | undefined
+  >(undefined);
+  const prefersDarkMode = useMediaQuery("(prefers-color-scheme: dark)");
+
+  const FhirIconCell = () => (
+    <IconButton size="small" className={classes.icon}>
+      <Icon
+        icon={IconNames.FLAME}
+        iconSize={15}
+        className={classes.iconFlame}
+      />
+    </IconButton>
+  );
+
+  const handleFhirIconClick = (index: number) => {
+    const primarykey = mapping?.primary_key_table;
+    if (primarykey) {
+      const indexPrimaryKey = exploredData.fields.indexOf(primarykey);
+      console.log(exploredData.rows[index]?.[indexPrimaryKey]);
+      setFhirInstance(patientJson);
+    }
+  };
+
+  return (
+    <Container className={classes.container} maxWidth="xl">
+      <Link
+        className={classes.link}
+        variant="h5"
+        color="textSecondary"
+        to={
+          attributeId
+            ? `/sources/${sourceId}/mappings/${mappingId}/attributes/${attributeId}`
+            : `/sources/${sourceId}/mappings/${mappingId}`
+        }
+        gutterBottom
+      >
+        <ArrowBack />
+        {t("back")}
+      </Link>
+      <TableContainer className={classes.table}>
+        <Table size="small">
+          <TableHead>
+            <TableRow className={classes.cellsTitle}>
+              <TableCell className={classes.cells}></TableCell>
+              {exploredData.fields.map((field) => (
+                <TableCell className={classes.cells} key={uuid()}>
+                  {field}
+                </TableCell>
+              ))}
+            </TableRow>
+          </TableHead>
+          <TableBody>
+            {exploredData.rows.map((columnData, index) => (
+              <TableRow hover key={uuid()} className={clsx(classes.rowBorder)}>
+                <TableCell
+                  className={clsx(classes.fhirIconCell, classes.cellsTitle)}
+                  onClick={() => handleFhirIconClick(index)}
+                >
+                  <FhirIconCell />
+                </TableCell>
+                {columnData.map((an, i) => (
+                  <TableCell className={classes.cells} key={i}>
+                    {an}
+                  </TableCell>
+                ))}
+              </TableRow>
+            ))}
+          </TableBody>
+          <TableFooter></TableFooter>
+        </Table>
+      </TableContainer>
+      {!fhirInstance && (
+        <div className={classes.texts}>
+          <Typography>
+            {t("clickOnAFireIcon")}{" "}
+            <Icon icon={IconNames.FLAME} className={classes.iconFlame} />{" "}
+            {t("inOrderToPreview")}
+          </Typography>
+        </div>
+      )}
+      {fhirInstance && (
+        <Container className={classes.jsonViewer} maxWidth="xl">
```

Comment:
I don't really like using a `Container` here, I brings some CSS logic with it that you prevent within your `jsonViewer` classeName, so what is the point ?

Reply:
you can't add a className to JsonViewer...

Reply:
Yes, I just suggest to use a div instead because I think you're overriding the implemented style of `Container`, but I may be mistaking. 
If changing `Container` by a `div` changes the style then don't bother to change it ;)